awx/main/db/profiled_pg/base.py

```diff
@@ -0,0 +1,101 @@
+import os
+import traceback
+import uuid
+
+from django.core.cache import cache
+from django.core.cache.backends.locmem import LocMemCache
+from django.db.backends.postgresql.base import DatabaseWrapper
+
+from awx.main.utils import memoize
+
+__loc__ = LocMemCache(str(uuid.uuid4()), {})
+__all__ = ['DatabaseWrapper']
+
+
+class RecordedQueryLog(object):
```

Comment:
I implemented this as a proxy _instead_ of as a subclass of `collections.deque` because `deque` isn't a python-based builtin, and I couldn't get the `append` override to work properly.  I might try to come up with a _cleaner_ way to do this.